### ISSUE TITLE
Make today dotter line in chart optional

### ISSUE THREAD

**Issue by adriaanvanrossum** [simpleanalytics/roadmap#504]
Some customers don't like the today part. We should make this optional and just hide the while today thing when disabled.

![image](https://user-images.githubusercontent.com/1079135/103642077-f4ecbc00-4f52-11eb-895f-edef20120619.png)

**Comment by RubenVanEldik** [comment]
Thank you for taking it up!

My main issue with the dotted line right now is that it goes to 0 for all future hours. This looks a bit weird and makes reading the number of visitors in the current hour quite hard. The dotted line might be fine if it just stopped at the current hour, instead of going to 0.

This also makes more sense since you're using an area-graph, and there is now also some colored area below the current hour and hour+1, which is technically incorrect. 🙃

**Comment by adriaanvanrossum** [comment]
So it's showing you the next hour? That's a bug! Let's fix that.

**Comment by RubenVanEldik** [comment]
Well, this is only a problem in the day view. It shows all hours of the day, which is fine, but the line going back to 0 makes it seem a bit abrupt.

![IMG_3F345696289A-1](https://user-images.githubusercontent.com/25854734/103649063-215a0580-4f5e-11eb-8e83-4bf510adb901.jpeg)

**Comment by adriaanvanrossum** [comment]
What would expect in this daily view situation @RubenVanEldik?

**Comment by RubenVanEldik** [comment]
The problem is that the dotted line goes to 0 for all future hours. This makes it especially hard to read if the number of view for the current hour are roughly half of last hour, since it looks like the dotted line goes in a linear line to 0. (while in reality it goes from 600 last hour to 300 this hour to 0 next hour.

I can think of 3 solutions: 

1. Use a bar chart. The number of visitors per hour are often not very related from one hour to the next (except if you have a big spike due to HN/Twitter etc.). Since there isn't really a trend it makes a bit more sense to use a bar chart.

2. Make the line from n-1 to n dotted and remove the line for n to >n.

![103649063-215a0580-4f5e-11eb-8e83-4bf510adb901](https://user-images.githubusercontent.com/25854734/105782397-c1f78000-5f74-11eb-998c-42394266ec75.jpeg)

3. Make the dotted line move forward at the same rate of the current hour. This kind of shows what would happen if the site keeps the same pace as the current hour and makes it even easier to see the number of views for the current hour. (this is an interesting approach, though I think it might lead to confusion for some users).

![103649063-215a0580-4f5e-11eb-8e83-4bf510adb901](https://user-images.githubusercontent.com/25854734/105782470-e6ebf300-5f74-11eb-8072-0270b53311d5.jpeg)

**Comment by RubenVanEldik** [comment]
My preference is for option 1, since its technically the best way to show a non-trend dataset.

**Comment by adriaanvanrossum** [comment]
Thanks! The bar chart will have "empty" hours as well, right?

**Comment by RubenVanEldik** [comment]
Yes, probably a striped/different colour/semi-transparant bar for the current hour and no bars for all future hours.

**Comment by amritnagi** [comment]
Not sure if it is possible but I think a nice thing to do would be to show a dot for the current hour value and have a pulse effect on it to show its “live” count (can mock something up later)
